client/structs/allochook.go

```diff
@@ -7,23 +7,31 @@ import (
 )
 
 // AllocHookResources contains data that is provided by AllocRunner Hooks for
-// consumption by TaskRunners
+// consumption by TaskRunners. This should be instantiated once in the
+// AllocRunner and then only accessed via getters and setters that hold the
+// lock.
 type AllocHookResources struct {
-	CSIMounts map[string]*csimanager.MountInfo
+	csiMounts map[string]*csimanager.MountInfo
 
 	mu sync.RWMutex
 }
 
+func NewAllocHookResources() *AllocHookResources {
+	return &AllocHookResources{
+		csiMounts: map[string]*csimanager.MountInfo{},
+	}
+}
+
 func (a *AllocHookResources) GetCSIMounts() map[string]*csimanager.MountInfo {
 	a.mu.RLock()
 	defer a.mu.RUnlock()
 
-	return a.CSIMounts
+	return a.csiMounts
 }
```

Comment:
What's the policy on modifying this map? Maybe add a comment here on what is expected of callers, or return a copy.

Reply:
Good call. It's probably safe to "guard" this with a comment because the volume hook is just reading that data and not writing it, but it's way to easy to break later unexpectedly. Returning a copy in a9b2ac844093863c731540caf57cb6cc8d7626fd